packages/pages-shared/__tests__/asset-server/handler.test.ts

```diff
@@ -0,0 +1,60 @@
+import { generateHandler } from "../../asset-server/handler";
+import { createMetadataObject } from "../../metadata-generator/createMetadataObject";
+import type { Metadata } from "../../asset-server/metadata";
+
+describe("asset-server handler", () => {
```

Comment:
@GregBrimble I think we should duplicate the existing private integration testing logic we have here in this unit test file. Thoughts?

Reply:
Would love to move most of that here, yes. The initial migration was a bit gnarly and I didn't want to unpack all the funky remote edge mode tests, so I just left them where they were, but yeah, if we can get the bulk of the relevant tests here beside the source, that would be amazing.

Reply:
> Would love to move

I'll create a ticket for us